File: C++/Niveau 1/8 - Répétitions conditionnées/Département de pédagogie.cpp
#include <iostream>
using namespace std;

int main(){

    bool aTrouve = false;
    int chiffreATrouver, nombreTentative = 0;

    cin >> chiffreATrouver;

    while (!aTrouve){
        int proposition;
        cin >> proposition;

        if(proposition < chiffreATrouver){
            cout << "c'est plus" << endl;
            nombreTentative++;
        }else if(proposition > chiffreATrouver){
            cout << "c'est moins" << endl;
            nombreTentative++;
        }else{
            aTrouve = true;
            nombreTentative++;
        }

    }

    cout << "Nombre d'essais nécessaires :"<< endl;
    cout << nombreTentative << endl;

}

File: Java/Niveau1/8- Répétitions conditionnées/Département de pédagogie.java
import algorea.Scanner;

class Main{
    static Scanner entrée = new Scanner(System.in);     

    public static void main(String[] args){
        int nombre = entrée.nextInt();
        int nombre_proposition = 0;
        int proposition = 1000;
        while(proposition != nombre){
           proposition = entrée.nextInt();
            if(proposition == nombre){
                nombre_proposition++;
                print("Nombre d'essais nécessaires :");
                print(nombre_proposition);
                break;
            }else if (proposition < nombre){
                print("c'est plus");
                nombre_proposition++;
            }else if (proposition > nombre){
                print("c'est moins");
                nombre_proposition++;
            }
        }
    }

    private static void print(Object object){
        System.out.println(object);
    }
}
